### PR TITLE
[KVS] Check NULL pointer for getPref_bin_new

### PR DESCRIPTION
**Pull request by pankore**
#### Problem
* Fix the hard fault caused by invalid read_bytes_size pointer.

#### Change overview
Add NULL pointer checker for getPref_bin_new

#### Testing
Tested commissioning and control

### CODE REVIEW

File: src/platform/Ameba/KeyValueStoreManagerImpl.cpp [hunk 1]
```diff
@@ -49,19 +49,22 @@ CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t
...
-        return CHIP_NO_ERROR;
-    case -6:
-        return CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
-    case -7:
-        return CHIP_ERROR_INVALID_ARGUMENT;
-    case -8:
-        return CHIP_ERROR_BUFFER_TOO_SMALL;
-    default:
-        break;
+        ret = getPref_bin_new(key, key, (uint8_t *) value, value_size, read_bytes_size);
```
Comment (Damian-Nordic) [inline src/platform/Ameba/KeyValueStoreManagerImpl.cpp:54]: I think it should still be possible to read the value even if `read_bytes_size` is nullptr. Just the actual size should not be reported.

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]

File: src/platform/Ameba/KeyValueStoreManagerImpl.cpp (hunk 1 above)
Comment (pankore) [reply src/platform/Ameba/KeyValueStoreManagerImpl.cpp:54]: This has been changed, actual size will not be reported if read_bytes_size is null.

**Review by Damian-Nordic — APPROVED** [review]

File: src/platform/Ameba/KeyValueStoreManagerImpl.cpp [hunk 2]
```diff
@@ -49,7 +49,16 @@ CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t
...
     }
 
-    ret = getPref_bin_new(key, key, (uint8_t *) value, value_size, read_bytes_size);
+    if (read_bytes_size)
+    {
+        ret = getPref_bin_new(key, key, (uint8_t *) value, value_size, read_bytes_size);
+    }
+    else
+    {
+        size_t * dummy_read_bytes_size = (size_t *) pvPortMalloc(sizeof(size_t));
```
Comment (andy31415) [inline src/platform/Ameba/KeyValueStoreManagerImpl.cpp:58]: Should we check the result from pvPortMalloc?

**Review by andy31415 — APPROVED** [review]

File: src/platform/Ameba/KeyValueStoreManagerImpl.cpp (hunk 2 above)
Comment (pankore) [reply src/platform/Ameba/KeyValueStoreManagerImpl.cpp:58]: done, thanks

**Pull request merged**